### DISCUSSION TITLE
MySQL support through socket connections for Cloud connections

### DISCUSSION THREAD

**Discussion by stx-chris** [directus/directus#7420] [Feature Requests]
Since v9 the former `DB_SOCKET` environment variable is not supported anymore. In order to connect from dockerized environments like Google Cloud Run securely to a Cloud SQL database, unix sockets are utilized as described [here](https://cloud.google.com/sql/docs/mysql/connect-run). They are named `/cloudsql/<myproject>:<myregion>:<myinstance>` and bound to the correct instance on bootup.

Whereas with PostgreSQL a socket connection can be communicated to the driver by setting the `DB_HOST` env variable, for MySQL the parameter to be set would be `socketPath`.

Knex does indeed support this alternative to `host` and `port` as specified [here](https://knexjs.org).

Can we have a `DB_SOCKET` variable that is sent to the MySQL driver if set? (P.S. Knex ignores `host` and `port` as soon as `socketPath` is available).

**Comment by rijkvanzanten** [comment]
We already pass on every `DB_*` environment variable to the knex config, so that part should be handled as is. 

The only thing we'd have to change is to not validate for DB_HOST etc to exist, when DB_SOCKET_PATH is provided, which is very similar to the logic in use for Postgres' DB_CONNECTION_STRING:

https://github.com/directus/directus/blob/f933663972c0788acc364f55a2787e8bcce5cba9/api/src/database/index.ts#L40-L46

Wanna open a PR for that @stx-chris? 🙂

**Comment by u12206050** [comment]
This seems to work correct?
I have now deployed to Cloud Run using the DB_SOCKET_PATH env variable.

**Reply by stx-chris** [reply]
In case of MySQL and the current codebase, you should be required to specifiy a DB_HOST entry as well (the fix is still pending). What did you put in DB_HOST?

**Reply by fczuardi** [reply]
you can put anything on the DB_HOST var since it will be ignored if DB_CONNECTION_STRING is passed, see https://github.com/directus/directus/blob/f933663972c0788acc364f55a2787e8bcce5cba9/api/src/database/index.ts#L54

**Reply by fczuardi** [reply]
what worked here for CloudRun + CloudSQL (pg) to work was a DB_CONNECTION_STRING in the following format (assuming user=postgres, password=12345, db=directus):

**DB_CONNECTION_STRING**=`postgresql://postgres:12345@directus?host=/cloudsql/gcloud-project-id:gcloud-region:cloudsql-id&sslmode=disable`

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
